sbnana/CAFAna/Core/SpectrumLoader.cxx

```diff
@@ -112,6 +112,54 @@ namespace ana
     fHistDefs.Clear();
   }
 
+  //----------------------------------------------------------------------
+  void SpectrumLoader::Select(unsigned int N)
+  {
+    if(fGone){
+      std::cerr << "Error: can only call Go() once on a SpectrumLoader" << std::endl;
+      abort();
+    }
+    fGone = true;
+
+    // Find all the unique cuts
+    //    std::set<Cut, CompareByID> cuts;
+    //    for(auto& shiftdef: fHistDefs)
+    //      for(auto& cutdef: shiftdef.second)
+    //        cuts.insert(cutdef.first);
+    //    for(const Cut& cut: cuts) fAllCuts.push_back(cut);
+
+    //    fLivetimeByCut.resize(fAllCuts.size());
+    //    fPOTByCut.resize(fAllCuts.size());
+
+
+    const int Nfiles = NFiles();
+
+    Progress* prog = 0;
+
+    caf::SRBranchRegistry::clear();
+
+    int fileIdx = -1;
+    while(TFile* f = GetNextFile()){
+      ++fileIdx;
+
+      if(Nfiles >= 0 && !prog) prog = new Progress(TString::Format("Filling %lu spectra from %d files matching '%s'", fHistDefs.TotalSize(), Nfiles, fWildcard.c_str()).Data());
+
+      HandleFile(f, Nfiles == 1 ? prog : 0, N);
```

Comment:
This is going to cause Select(N) to handle the Nth event in every file, which is probably not what you want.